### ISSUE TITLE
Linux target: execute command not properly escaped

### ISSUE THREAD

**Issue by FWest98** [microsoft/DockerTools#245]
My project path has spaces in it, and this seems to be handled incorrectly at the moment. Building the images, starting the container and mounting the volumes all works perfectly fine. However, the command to actually run seems to be broken. When looking at the Container Tools output, I see the following command being executed:
`docker exec -i ... <container-ID> sh -c ""dotnet" <nuget paths> "/path/Some File.dll" | tee /dev/console"`
As you might be able to see, this will fail. All 'inner' strings are not properly escaped: the effect is that the container tries to execute `dotnet` with only a partial command. We can see this by stripping this to the bare minimum:
`docker exec -i ... <container-ID> sh -c ""cat" <leave paths> "/path/Some File.dll" <removed>"`
This will result in: `cat: /path/Some: No such file or directory`.

This is currently quite annoying and prevents me from properly using VS with Linux containers. For Windows containers, it's all working fine.
Running VS 16.5.0 with nuget package version 1.9.10.

**Comment by FWest98** [comment]
Note that this bug only emerges when actually using paths with spaces; when not using spaces the quotes are still incorrect, but it reduces to a valid command.

**Comment by pratiksanglikar** [comment]
Hi @FWest98 , I don't think the quotes are incorrect, they are in fact matching.
It seems that the problem is with docker, they don't seem to handle spaces very well.

I have distilled this problem to a simple problem, you can try executing following docker exec command on Powershell.

```
PS C:\Users\username> docker exec -i <containerId> sh -c ""echo" "/app/bin/Debug/netcoreapp3.1/space test.dll""
PS C:\Users\username>/app/bin/Debug/netcoreapp3.1/space
```

Note that how the `docker exec` command has ignored anything that comes after a space.
I'm still investigating if we can escape the spaces somehow, but my gut says we'll have to take up this issue to Docker for resolution.

**Comment by FWest98** [comment]
If you have
`sh -c ""echo" "/path""`
then the first two `""` will cancel, leaving
`sh -c echo " "/path""`
the the second two `" "` will cancel:
`sh -c echo /path""`
and then the third pair will cancel, leaving the path with spaces unescaped.

When I try to escape the outer command with `''` instead, it works fine:
`docker exec -i <containerId> sh -c '"echo" "/path "'`

**Comment by pratiksanglikar** [comment]
Using single quotes instead of double quotes won't help either.
I just tried -
```
 docker exec -i a9 sh -c '"echo" "/app/bin/Debug/netcoreapp3.1/space test.dll"'                                                                                       /app/bin/Debug/netcoreapp3.1/space
```

**Comment by pratiksanglikar** [comment]
Another quick test I did was to check what happens when any of the paths in your dockerfile has spaces in it.
Simple copy command -
`COPY ["space test/space test.csproj", "space test/"]`
would fail to copy on Linux containers.

**Comment by FWest98** [comment]
When trying it in the console it seems to be dependent on the exact shell I'm using. I tried various commands in various shells, yielding:
- As in the issue description with all "":
`docker run --rm -it busybox sh -c ""echo" "foo bar""`
cmd: `foo`; powershell: `foo`; bash (WSL): `foo`
- With single quotes, as I mentioned before:
`docker run --rm -it busybox sh -c '"echo" "foo bar"'`
cmd: syntax error; powershell: `foo`; bash: `foo bar`
- With escaping inner quotes (using `` ` `` in powershell, and `\` in other shells)
`docker run --rm -it busybox sh -c "\"echo\" \"foo bar\""`
cmd: `foo bar`; powershell: `foo`; bash: `foo bar`

So this is how I got to the single quotes thing. Oddly enough, the escaping does not work in Powershell, while a simple ``echo "`"echo`" `"foo bar`""`` does properly print `"echo" "foo bar"`... Anyway, the results above indicate that docker seems to function properly. In my experience, I also do not have any problems with a copy with spaces in it; both targeting Windows and Linux containers. I'm running VS 16.5.0 and plugin version 1.9.10.
